### PR TITLE
makefile: Make misc improvements

### PR DESCRIPTION
**Pull request by euank**
* Add test target and re-use in travis
* Add timeout test target
* Build binary into ./bin folder
* Phony everything

I thought I had made a PR with this, but nope, it was just sitting around in my branch for the last week.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+
 all: push
 
 # See node-problem-detector.yaml for the version currently running-- bump this ahead before rebuilding!
 TAG = v0.2
 
 PROJ = google_containers
 
-node-problem-detector: node_problem_detector.go
+PKG_SOURCES := $(shell find pkg -name '*.go')
```
Comment (Random-Liu) [inline Makefile]: Why not include go files in `vendor`? We should also rebuild if `vendor` is updated, right?

File: Makefile [hunk 2]
```diff
@@ -1,18 +1,27 @@
+.PHONY: all phony container push clean node-problem-detector
```
Comment (Random-Liu) [inline Makefile]: What does the `phony` mean here?

File: Makefile (hunk 1 above)
Comment (euank) [inline Makefile]: It's a premature optimization. Managing the timestamps for that many files is a bit slower and having to do `make clean` after a vendor update doesn't seem like that big of a deal.

Comment (Random-Liu) [inline Makefile]: Hm, if so. Let's not check `vendor` for now. :)

**Comment by Random-Liu** [comment]
LGTM. Will merge this after test passed.

**Pull request merged**